### ISSUE TITLE
Diffusion Policy Success Comparison in Robomimic

### ISSUE THREAD

**Issue by RochMollero** [irom-lab/dppo#41]
Hello,

I've read the DPPO paper, which is a great piece of research & bibliography. I am interested in evaluating your code base for improving diffusion policies, especially on pixel inputs, but I'm confused about the results in the paper:

Figure 7 shows that the best DPPO makes the perf goes from
 - 0.4 to 1.0 on square - state
 - 0.25 to ~0.9 on transport - state
 - 0.18 to ~0.9 on square - pixel
 - 0.35 to ~0.9 on transport - pixel

However in the original Diffusion Policy paper, the average performance of the raw IL diffusion algorithm (without RL) (Table I and II here : https://diffusion-policy.cs.columbia.edu/diffusion_policy_2023.pdf)

 - 0.93 or 0.82 on square - state  depending on proficient or multi human
 - 0.84 or 0.46 on transport - state  depending on proficient or multi human
 - 0.92 or 0.84 on square - pixel  depending on proficient or multi human
 - 0.93 or 0.69 on transport - pixel  depending on proficient or multi human

So I have three questions:
 - Why are you having such low success at the start of the RL ? is it because of the exploration noise only, or because the pretrained policy was not as good as the one in the Diffusion Policy paper ? If so Why ?

 - does the RL really brings better results at the end, than vanilla Diffusion Policy on proficient human ? This seem unclear from the data. This is particularly important since you claim "To our knowledge, DPPO is the first RL algorithm to solve Transport from either state or pixel input to high (>50%) success rates" : but Diffusion Policy is at  80%+ on average with proficient data

 - Also in Figure 7, what does mean the x label "Environement step" ? Is it actually training steps ?

Regards,
Roch

**Comment by allenzren** [comment]
Hey @RochMollero,

We purposely used worse IL policy (smaller in size or undertrained) so there is more room for RL policy to improve. We would need this to benchmark DPPO against other RL algorithms.

To our knowledge, DPPO is the first RL algorithm to solve Transport from either state or pixel input to high (>50%) success rates" : but Diffusion Policy is at 80%+ on average with proficient data ---> Diffusion Policy is not a RL algorithm. We are just interested in whether DPPO as a RL algorithm improves the performance. All the other RL algorithms cannot optimize the policy at all.

Environment steps mean the number of steps sampled from the training environment, not the number of gradient step.

**Comment by RochMollero** [comment]
Thanks for the answer, that makes complete sense. But then have you a definitIve answer to whether 

"The Best Diffusion Policy + DPPO" outperforms "The Best Diffusion Policy" in your opinion / experience ?

If possible that would be great to have the same answer regarding this paper too : https://arxiv.org/pdf/2407.16677v1 (which the author seem to be the same than the DPPO paper) : does it dramatically improve on the best Diffusion Policy or not ?


Regards
